### PR TITLE
Scholarship field

### PR DESCRIPTION
**Pull request by clareconstantine**
Add scholarship field to applications and detail view (CSVs, cohort and quick view to follow)

[spec](https://docs.google.com/document/d/17GOvnMCPohlSqbuXr2k3EWRqOfngT81_wkHe6B6UY20/edit#)

### CODE REVIEW

File: dashboard/db/schema.rb [hunk 1]
```diff
@@ -1067,7 +1067,7 @@
     t.index ["user_id", "script_id", "level_id"], name: "index_puzzle_ratings_on_user_id_and_script_id_and_level_id", unique: true, using: :btree
```
Comment (clareconstantine) [inline dashboard/db/schema.rb]: I didn't intentionally change this file - fixing now

File: apps/src/code-studio/pd/application_dashboard/detail_view_contents.jsx [hunk 2]
```diff
@@ -369,6 +384,31 @@ export class DetailViewContents extends React.Component {
     return this.state.regional_partner_name;
   };
 
+  renderScholarshipStatusAnswer = () => {
+    if (this.state.editing && this.props.isWorkshopAdmin) {
+      return (
+        <FormGroup>
+        <Select
```
Comment (mehalshah) [inline apps/src/code-studio/pd/application_dashboard/detail_view_contents.jsx]: I think this should be indented two spaces from the above line

File: apps/src/code-studio/pd/application_dashboard/detail_view_contents.jsx [hunk 3]
```diff
@@ -369,6 +384,31 @@ export class DetailViewContents extends React.Component {
...
+          value={this.state.scholarship_status}
+          onChange={this.handleScholarshipStatusChange}
+          options={this.props.applicationData.scholarship_dropdown_options}
+        />
+      </FormGroup>
```
Comment (mehalshah) [inline apps/src/code-studio/pd/application_dashboard/detail_view_contents.jsx]: This should also be indented

File: dashboard/app/serializers/api/v1/pd/application_serializer.rb [hunk 4]
```diff
@@ -139,4 +141,16 @@ def school_stats
       two_or_more_races_percent: percent_string(stats.student_tr_count, stats.students_total)
     }
   end
+
+  def scholarship_status
+    object.try(:scholarship_status)
+  end
+
+  def scholarship_dropdown_options
```
Comment (mehalshah) [inline dashboard/app/serializers/api/v1/pd/application_serializer.rb]: I don't think this list of options belongs in the serializer. This is the same for every application, and shouldn't be something we serialize every single time.

Instead this map should be in Shared Constants

**Comment by clareconstantine** [comment]
[Scholarship field for teacher applications](https://trello.com/c/blNduC9U/757-scholarship-field-for-teacher-applications)

**Pull request merged**